### DISCUSSION TITLE
`CanExecute` and data binding

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#12202] [Q&A]
Hi,

I read https://learn.microsoft.com/en-us/dotnet/maui/xaml/fundamentals/mvvm?view=net-maui-7.0#commanding and I see that in the sample there is the following method call `((Command)DeleteCharCommand).ChangeCanExecute();`.

Isn't it possible to avoid doing it explicitly and just do this `ChangeCanExecute` in a declarative way via (data) binding?

Thank you

**Comment by BioTurboNick** [comment]
Determining whether a command can be executed or not may be more complex than binding to a single property. It may depend on the program state, what fields are populated, what other commands might currently be executing, etc.

**Reply by MartyIX** [reply]
The data binding property does not need to be trivial.

**Reply by BioTurboNick** [reply]
This is no different than implementing `INotifyPropertyChanged`. Something has to check for the change to occur and raise the event. And if it's a read-only property, you need to manually indicate when it changes elsewhere. So I'm not sure what you're asking for?
